src/librustc_session/options.rs

```diff
@@ -715,7 +715,7 @@ options! {CodegenOptions, CodegenSetter, basic_codegen_options,
         "a single extra argument to append to the linker invocation (can be used several times)"),
     link_args: Vec<String> = (Vec::new(), parse_list, [UNTRACKED],
         "extra arguments to append to the linker invocation (space separated)"),
-    link_dead_code: bool = (false, parse_bool, [UNTRACKED],
+    link_dead_code: Option<bool> = (None, parse_opt_bool, [UNTRACKED],
```

Comment:
I don't understand why this changed from a `bool` to an `Option<bool>`. Can you elaborate a bit?

Reply:
IFF -Zinstrument-coverage is enabled, I want to default -Clink-dead-code=on, but allow the user to force -Clink-dead-code=off.

The coverage analysis still works, but with link-dead-code off, the dead code has no counters instead of zeros (see the example at the top of this PR; with link-dead-code off, the red highlight goes away and the left-hand column of counts is blank for the dead function. That might be desired in some cases, but shouldn't be the default in this case.